### DISCUSSION TITLE
What does "Enhance the Enter key" mean?

### DISCUSSION THREAD

**Discussion by rbanks88** [vslinko/obsidian-outliner#98] [Q&A] [answered]
The Outliner plugin has a setting to make the enter key behave like other outliners. What does that mean, exactly? 

In OmniOutliner, when I hit enter, it creates a new row, instead of splitting the current row at the cursor position. But the Outliner plugin just splits the line, whether or not "Enhance the Enter key" activated. Thanks for explaining this!

Regards,
Russell

**Comment by vslinko** [comment, marked as answer]
This setting means two features that inspired by default behaviour of Roam Research and Workflowy.

1. Enter should outdent if it's an empty list item. For example:
    
    ```
    For this list (vertical bar is the cursor):
    
    - one
        - two
            - |
    
    Pressing Enter will outdent list item:
    
    - one
        - two
        - |
    ```

2. Enter creates new line on children level:

    ```
    For this list (vertical bar is the cursor):
    
    - one
        - two|
            - three
    
    Pressing Enter will create list item on children level:
    
    - one
        - two
            - |
            - three
    ```

3. Shift-Enter creates a new note line.
    
    ```
    For this list (vertical bar is the cursor):
    
    - one
        - two|
    
    Pressing Shift-Enter will create a note line:
    
    - one
        - two
            |
    ```
